webapp/channels/src/components/user_settings/general/user_settings_general.tsx

```diff
@@ -472,6 +527,27 @@ export class UserSettingsGeneralTab extends PureComponent<Props, State> {
         }
     };
 
+    updateSelectAttribute = (selectedOption: OnChangeValue<SelectOption, boolean>, action: ActionMeta<SelectOption>, fieldID: string) => {
+        const attributeValues = {...this.state.customAttributeValues};
+
+        if (!selectedOption) {
+            attributeValues[fieldID] = '';
+        } else if (Array.isArray(selectedOption)) {
+            // Handle multi-select
+            const values = selectedOption.filter((option): option is SelectOption => Boolean(option && 'value' in option && option.value)).map((option) => option.value);
```

Comment:
1/5, non-blocker.
`Object.forOwn` would be better like down in https://github.com/mattermost/mattermost/pull/30317/files#diff-201fb315dd10cfac83b7c4dcb38961bda077838993b3892c2ef1cb653ea62ea5R1523

Reply:
I think you meant this on the next check. 
`else if ('value' in selectedOption)`.
I tried using `Object.forOwn`, but that caused an error on `attributeValues[fieldID] = selectedOption.value`. It still thinks `selectedOption` may be a `MultiValue`. I tried checking for `!Array.isArray()`, etc. Not sure it will work here.

Reply:
Yeah this is a tricky one. I wonder if splitting into dedicated single-select and multi-select handlers would be simpler. (0/5)